rfcs/0145-doc-strings.md

```diff
@@ -119,19 +117,157 @@ In the following we give a comprehensive overview to our decision that we've mad
 
+Accurate and usable Lambda documentation reqires some more details to be specified:
 
+- **If a lambda is assigned to an `attribute name`, then the `attribute name`-documentation is also the `lambda value` documentation**.
+    - In case both places have doc-comments, the one closer to the lambda has higher precedence.
+      
+- In case of curried functions all partial functions can share the same placement with the outermost lambda.
 
-# Decisions
+> Note: Research of the RFC Sheperds Team showed That this allows for intuitive placements like are already done in nixpkgs. 
+
+### Lambda Examples
+
+```nix
+/**Doc for anonymous lambda function*/
+↓
+x: x;
+
+# doc -> Lambda Value '(x: x)'
+```
+
+```nix
+/**Doc for assigned lambda function*/
+↓
+assigned = x: x;
+```
+
+```nix
+/**Doc B*/
+assigned = /**Doc A*/x: x;
+
+# Lambda Documentation is 'Doc A' because it is directly next to the documentable lambda.
+```
+
+### Attribute Examples
+
+```nix
+{
+    /** Documentation for 'foo' */
+    ↓ 
+    foo = 1;
+}
+# doc -> Attribute Name foo
+```
+
+```nix
+{
+    /** Documentation for 'bar' */
+    ↓
+    ${let name = "bar"; in name} = 2;
+}
+# Interpolated attribute names
+```
+
+```nix
+{
+    /** Documentation for 'bar' */
+    ↓
+    ${let name = "bar"; in name} = 2;
+}
+# Interpolated attribute names
+```
+
+```nix
+setWithMappedNamed =
+    lib.mapAttrs' (name: value: {
+      /**Undocumentable attribute name*/
+              ↓
+      name = "${n + "foo"}";
+      inherit value;
+    })
+    {
+      a = 1;
+      b = 2;
+    };
+# Dynamically constructed attribute names cannot be documented. (constructed in a function call like above)
+``` 
+
+### Examples when the lambda is not directly assigned
+
+```nix
+{
+    /**This documents (1), NOT (2) */
+    ↓1    ↓2
+    foo = map (x: x);
+          ^~~~~~~~~~~ Evaluates to a specialisation of map, which is a partially applied lambda and defined elsewhere.
+          
+# IMPORTANT: Documentation of the specialized map function can still be done as shown. But is related to attrName 'foo' (1)
+}
+```
+
+```nix
+{
+    /**This documents (1), NOT (2) */
+    ↓1                           ↓2
+    foo = mapAttrs (name: value: (x: x) ) {a=1; b=2;};
+    # => 
+    # foo = {
+    #   a = x: x;
+    #   b = x: x; 
+    # }
+# Doesn't document the inner anonymous lambda function.
+}
+```
+
+### Partially applied lambda
+
+All partial functions can share the same placement with the outermost lambda.
+
+```nix
+/**Generic doc for all partial lambda functions too*/
+↓1 ↓2          ↓3
+x: ({y, ...}:  z: x + y) * z;
+
+# 1 -> Lambda Value `x: ({y, ...}: z: x + y) * z` countApplied: 0
+# 2 -> Lambda Value `({y, ...}: z: x + y) * z`    countApplied: 1
+# 3 -> Lambda Value `z: x + y) * z`               countApplied: 2
+```
 
+```nix
+{
+    /** incrMap. Adds one to every item. */
+    ↓    
+    incrMap = map (x: x + 1);
```

Comment:
`map` is a primop isn't it? why it is defined elsewhere? I understand what you mean here, but I think we should pick up a curried lambda for this example, `map` might be confusing for readers.